tsl/src/backup.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+#include <postgres.h>
+#include <funcapi.h>
+#include <utils/builtins.h>
+#include <utils/pg_lsn.h>
+#include <utils/guc.h>
+#include <access/xlog_internal.h>
+#include <access/xlog.h>
+#include <access/xact.h>
+#include <catalog/pg_foreign_server.h>
+#include <storage/lmgr.h>
+#include <miscadmin.h>
+
+#include "errors.h"
+#include "guc.h"
+#include "catalog.h"
+#include "debug_wait.h"
+#include "dist_util.h"
+#include "remote/dist_commands.h"
+#include "backup.h"
+
+enum
+{
+	Anum_restore_point_node_name = 1,
+	Anum_restore_point_node_type,
+	Anum_restore_point_lsn,
+	_Anum_restore_point_max
+};
+
+static Datum
+create_restore_point_datum(TupleDesc tupdesc, const char *node_name, XLogRecPtr lsn)
+{
+	Datum values[_Anum_restore_point_max] = { 0 };
+	bool nulls[_Anum_restore_point_max] = { false };
+	HeapTuple tuple;
+
+	tupdesc = BlessTupleDesc(tupdesc);
+	if (node_name == NULL)
+	{
+		nulls[AttrNumberGetAttrOffset(Anum_restore_point_node_name)] = true;
+		values[AttrNumberGetAttrOffset(Anum_restore_point_node_type)] =
+			CStringGetTextDatum("access_node");
+	}
+	else
+	{
+		values[AttrNumberGetAttrOffset(Anum_restore_point_node_name)] = CStringGetDatum(node_name);
+		values[AttrNumberGetAttrOffset(Anum_restore_point_node_type)] =
+			CStringGetTextDatum("data_node");
+	}
+	values[AttrNumberGetAttrOffset(Anum_restore_point_lsn)] = LSNGetDatum(lsn);
+	tuple = heap_form_tuple(tupdesc, values, nulls);
+	return HeapTupleGetDatum(tuple);
+}
+
+Datum
+create_distributed_restore_point(PG_FUNCTION_ARGS)
+{
+	const char *name = TextDatumGetCString(PG_GETARG_DATUM(0));
+	DistCmdResult *result_cmd;
+	FuncCallContext *funcctx;
+	XLogRecPtr lsn;
+
+	if (SRF_IS_FIRSTCALL())
+	{
+		int name_len = strlen(name);
+		MemoryContext oldctx;
+		TupleDesc tupdesc;
+		char *sql;
+
+		if (name_len >= MAXFNAMELEN)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("restore point name is too long"),
+					 errhint("maximum length is %d, while provided name has %d chars",
```

Comment:
I think this sounds more like "detail" than "hint". A hint should be a concrete suggestion how to avoid the error, e.g., "Use a name with a maximum length of %d.". You can of course reformulate like that if you prefer a hint (often it is actually more helpful to tell users what to do than pointing out the details of why their command failed). It is a bit pointless, IMO, to point out the length of the restore point name since it was suggested by the user, so the hint message above might be more suitable.

If you prefer to keep the detail message, then tweak to conform to error guidelines:

```suggestion
					 errdetail("Maximum length is %d, while provided name has %d chars.",
```